### DISCUSSION TITLE
Licensing a Project

### DISCUSSION THREAD

**Discussion by jayasankar1595** [community/community#77157] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

In the development of my project, which is currently MIT licensed, I am considering accepting a contribution which is a few lines of code. The code that is being contributed is licensed under GPL v2.0. My specific concern is whether incorporating this GPL v2.0 licensed code into my MIT licensed repository necessitates a change in the overall license of my project to GPL v2.0 as per the license terms.
 
I would greatly appreciate your guidance on this matter. Your insights will be invaluable in ensuring that I adhere to the appropriate licensing requirements. Thank you in advance!

**Comment by MantasTech** [comment]
Hey there!

Mixing GPL code with an MIT project can be tricky. GPL v2.0 is a copyleft license, which means if you include any GPL'd code, your whole project needs to be distributed under GPL v2.0 as well. So yeah, if you incorporate that GPL v2.0 code into your project, you'd typically have to relicense your entire project to GPL v2.0.

Hope that helps

**Comment by bkdotcom** [comment]
Somewhat related..   I've added some "contains code from <list of other repos>" text to my LICENSE  
github now displays my license as "other" instead of "MIT License"   
how can I add some info and maintain  the "MIT License"  "label" ?
